electrumx/server/block_processor.py

```diff
@@ -434,20 +446,28 @@ def advance_txs(self, txs, is_unspendable):
                     continue
                 cache_value = spend_utxo(txin.prev_hash, txin.prev_idx)
                 undo_info_append(cache_value)
-                append_hashX(cache_value[:-15])
+                append_hashX(cache_value[:-TX_NUMB_LEN-TX_VALUE_LEN-IS_STAKE_FLAG_LEN])
 
             # Add the new UTXOs
+            staking_idx = -1
             for idx, txout in enumerate(tx.outputs):
                 # Ignore unspendable outputs
                 if is_unspendable(txout.pk_script):
+                    if len(txout.pk_script) == STAKING_HEADER_LENGHT:
+                        reversed_tx_hash = bytearray(tx_hash)
+                        reversed_tx_hash.reverse()
+                        if self.daemon.request_staking(reversed_tx_hash.hex()) is None:
```

Comment:
This call is quite costly, as it requires a database lookup at the node. I would perform an additional in-place check of the following things before calling the node:
- `txout.pk_script[:4] == b'\x6a\x04\x53\x44'`
- `0 < staking_idx < len(tx.outputs)`

Reply:
fixed